core/include/array/array_sorted_read_state.h

```diff
@@ -201,17 +202,14 @@ class ArraySortedReadState {
   /** AIO counter. */
   int aio_cnt_;
 
-  /** The AIO mutex conditions (one for each buffer). */
-  pthread_cond_t aio_cond_[2];
+  /** Condition variables used in AIO. */
+  std::condition_variable aio_cv_[2];
```

Comment:
Recall that this is the **sorted** case, which means that the user requires the cells to be sorted in the query subarray in row- or column-major, whereas the cells may be stored differently in TileDB. Hence, the CPU intensive task accounts for re-arranging properly the cells into the user's buffers, after the AIO has returned.